tools/unsupervised_dataset/Crawler/CrawlerLog.py

```diff
@@ -0,0 +1,71 @@
+import datetime
+import csv
+import os
+
+class CrawlerLog:
+    """ The CrawlerLog keeps track of which websites were explored, how many queries were found,
+        and creates a CSV with all the queries. It also logs any errors encountered. The log is
+        saved into Logs subdirectory with name based on start time. Queries are saved into Queries
+        subdirectory.
+    """
+    
+    def __init__(self):
+        """ Initializes crawler log to keep track of crawler progress and instantiates
+            instance variables.
+        """
+        
+        self.startTime = datetime.datetime.now().strftime("%Y-%m-%d-%H:%M")
+        
+        # Create directory for logs if it does not already exists
+        try:
+            os.mkdir("Logs")
+        except FileExistsError as e:
+            pass
+        
+        self.log = open("Logs/log-" + self.startTime + ".txt", "a")
+        self.log.write("Beginning crawl at time {0}.\n".format(self.startTime))
+        
+        try:
+            os.mkdir("Queries")
+        except FileExistsError as e:
+            pass
```

Comment:
I think we need to handle this case more carefully. Imagine people mistakenly run the tool with an old path, it may unexpectedly overwrite some existing crawled data?